lib/importers/revision_score_importer.rb

```diff
@@ -69,38 +58,37 @@ def update_previous_revision_scores
   ##################
   private
 
-  # The top-level key representing the wiki in LiftWing data
-  def wiki_key
-    # This assumes the project is Wikipedia, which is true for all wikis with the articlequality
-    # or the language is nil, which is the case for Wikidata.
-    @wiki_key ||= "#{@wiki.language || @wiki.project}wiki"
-  end
-
-  def model_key
-    @model_key ||= @wiki.project == 'wikidata' ? 'itemquality' : 'articlequality'
-  end
-
   def get_and_save_scores(rev_batch)
-    scores_data = @lift_wing_api.get_revision_data rev_batch.map(&:mw_rev_id)
-    scores = scores_data.dig(wiki_key, 'scores') || {}
+    scores = @lift_wing_api.get_revision_data rev_batch.map(&:mw_rev_id)
     save_scores(scores)
   end
 
   def get_and_save_previous_scores(rev_batch)
     parent_revisions = get_parent_revisions(rev_batch)
     return unless parent_revisions&.any?
-    parent_quality_data = @lift_wing_api.get_revision_data parent_revisions.values.map(&:to_i)
-    scores = parent_quality_data.dig(wiki_key, 'scores') || {}
+    scores = @lift_wing_api.get_revision_data parent_revisions.values.map(&:to_i)
     save_parent_scores(parent_revisions, scores)
   end
 
+  def save_scores(scores)
+    scores.each do |mw_rev_id, score|
+      revision = Revision.find_by(mw_rev_id: mw_rev_id.to_i, wiki_id: @wiki.id)
+      next unless revision
+      revision.wp10 = score.dig('wp10')
+      revision.features = score.dig('features')
+      # only modify the existing deleted value if revision was deleted
+      revision.deleted = true if score.dig('deleted')
```

Comment:
The spec modified in c04587283aa60c5f66a2b495b737bc704b8426de was to be sure that `revision.deleted` is not updated if `score.dig('deleted')` isn't set to true. If `score.dig('deleted')` is nil (because `score` is the empty hash {}), then `revision.deleted` is not updated.